src/ssl.c

```diff
@@ -54957,15 +54956,45 @@ int wolfSSL_CONF_cmd(WOLFSSL_CONF_CTX* cctx, const char* cmd, const char* value)
     WOLFSSL_BIO *wolfSSL_BIO_new_connect(const char *str)
     {
         WOLFSSL_BIO *bio;
+        const char* port;
         WOLFSSL_ENTER("wolfSSL_BIO_new_connect");
         bio = wolfSSL_BIO_new(wolfSSL_BIO_s_socket());
         if (bio) {
-            bio->ip = str;
+            port = XSTRSTR(str, ":");
```

Comment:
Previous code was able to keep a pointer to the incoming string, but I am not sure it was safe to assume that point would continue to be valid, so the malloc/copy is probably justified. Just wanted to check and make sure.

Reply:
OpenSSL also copies in the string. Its not safe to assume that the string will be kept around.